docs/source/guide/shadows-1-intro.md

```diff
@@ -0,0 +1,337 @@
+    k_calibration=1,
+    locality=2,
+    qubits=qubits,
+    executor=noisy_executor,
+    num_total_measurements_calibration=5000,
+)
+f_est
+```
+
+the varible `locality` is the maximum number of qubits that our interested operators acting on. e.g. if our interested operator is a sequence of two point correlation function $\{\langle Z_iZ_{i+1}\rangle\}_{0\leq i\leq n-1}$, then `locality` = 2. Noted here, one could also split the calibration process into two stage:
+
+01. `shadow_quantum_processing`
+   - Outcome: Get quantum measurement result of the calibration circuit $|0\rangle^{\otimes n}$ `zero_state_shadow_outcomes`.
+
+02. `pauli_twirling_calibration`
+   - Outcome: A dictionary of `calibration_results`.
+For more details, please refer to the see [rshadow tutorial for cirq](../examples/rshadows_tutorial.md)
+
+### 1. Quantum Processing
+In this step, we obtain classical shadow snapshots (before apply the invert channel) from the input state
+
+#### 1.1 Add Rotation Gate and Meausure the Rotated State in Computational Basis
+At present, the implementation supports random Pauli measurement. This is equivalent to randomly sampling $U$ from the local Clifford group $Cl_2^n$, followed by a $Z$-basis measurement (see [shadow tutorial for cirq](../examples/shadows_tutorial.md) for clear explaination).
+
+#### 1.2 Get the Classical Shadows
+One can obtain the list of measurement results of local Pauli measurement in terms of bitstrings, and the related Pauli-basis measured in terms of strings by calling the `shadow_quantum_processing` function with given the total number of snapshots `num_total_measurements_shadow`:
+
+
+```{code-cell} ipython3
+shadow_measurement_output = shadow_quantum_processing(
+    circuit,
+    noisy_executor,
+    num_total_measurements_shadow=5000,
+)
+```
+                                                                     
+
+for example, print out one of those measurement outcome and meausurement basis:
+
+
+```{code-cell} ipython3
+print("one snapshot measurement reslut = ", shadow_measurement_output[0][0])
+print("one snapshot measurement basis = ", shadow_measurement_output[1][0])
+```
+
+### 2. Classical Post-Processing
+In this step, we estimate our interested object by classical shadow protocol. 
+
+#### 2.1 Example: Operator Expectation Value Esitimation
+For example, if we want to estimate the expectation value of a sequence of two point correlation functions $\{\langle Z_iZ_{i+1}\rangle\}_{0\leq i\leq n-1}$, we will define the operators by Mitiq Paulistrings:
+
+```{code-cell} ipython3
+from mitiq import PauliString
+
+two_pt_correlations = [
+    PauliString("ZZ", support=(i, i + 1), coeff=1)
+    for i in range(0, num_qubits - 1)
+]
+for i in range(0, num_qubits - 1):
+    print(two_pt_correlations[i]._pauli)
+```
+
+One can estimation the expectation value of the observables with the previous classical shadows. One can get the estimation of expectation values without/with calibration:
```

Comment:
The corresponding expectation values can be estimated (with and without calibration) as shown in the next code cell.